backend/sirius-web-interpreter/src/main/java/org/eclipse/sirius/web/interpreter/AQLInterpreter.java

```diff
@@ -47,6 +53,29 @@
  */
 public class AQLInterpreter {
 
+    /**
+     * {@link CrossReferenceProvider} implementation which simply looks for an existing {@link ECrossReferenceAdapter}
+     * already attached to the source {@link EObject}. Returns an empty set of references without any error if no
+     * {@link ECrossReferenceAdapter} is found.
+     */
+    private static final class SimpleCrossReferenceProvider implements CrossReferenceProvider {
+        @Override
+        public Collection<Setting> getInverseReferences(EObject self) {
+            if (self != null) {
+                // @formatter:off
+                var xref = self.eAdapters().stream()
+                               .filter(ECrossReferenceAdapter.class::isInstance)
+                               .map(ECrossReferenceAdapter.class::cast)
+                               .findFirst();
+                // @formatter:on
+                if (xref.isPresent()) {
+                    return xref.get().getInverseReferences(self);
+                }
+            }
+            return Collections.emptySet();
+        }
+    }
```

Comment:
Could you make this a regular class and move it next to the AQLInterpreterFactory. Some users want to heavily customize the AQLInterpreter and thus they would need to customize that too. It would be better to see it as an argument in the constructor.

Reply:
I'll do that, but for now I just wanted to get these patches in proper PRs. I'll update them (with missing tests) afterwards, and take remarks into account then.